src/AppBundle/Entity/User.php

```diff
@@ -258,8 +258,9 @@ class User implements UserInterface
      */
     public function __construct()
     {
-        $this->clients = new \Doctrine\Common\Collections\ArrayCollection();
+        $this->clients = new ArrayCollection();
         $this->password = '';
+        $this->teams = new ArrayCollection();
```

Comment:
This constructor code always creates an empty object. This might be one reason why prevous attempts at doctrine persistence wasnt working...? A constructor should be capable of creating new and existing User objects: new User(); or new User($clients, $teams);
what about something like this: 
     public function __construct($clients = [], $teams = [])
      {
          $this->clients = new ArrayCollection($clients);
          $this->teams = new ArrayCollection($teams);
          ...
By specifying rules in the entity, it keeps business logic in the backend and *anything* that creates a User needs to adhere to the contract that the object specifies. In this case, to create a user I expect a list of clients and teams (although they can be empty)